Terminal.Gui/Configuration/SettingsScope.cs

```diff
@@ -118,6 +118,9 @@ public class SettingsScope : Scope<SettingsScope>
             return this;
         }
 
+        // BUG: Not trim-compatible
+        // Not a bug, per se, but it's easily fixable by just loading the file.
+        // Defaults can just be field initializers for involved types.
```

Comment:
I really like having a built-in config file for defaults. I hope there's a trim compatible way of doing so.

Reply:
You just initialize the variables in code. Same effect, no embedded text.

Reply:
To clarify:

An embedded text file is essentially just a more convoluted string constant, except that it's not a string constant or it would be safe. It's just some anonymous data with no references.

The trim analyzer doesn't understand that the method call used to retrieve it is what "references" it.

Regardless, it's still effectively a hard-coded default value for everything, making it functionally equivalent to Properties/fields having initializers or being initialized in a constructor.

Therefore, the only functional difference on our end is that you do have to make sure a change in one is reflected in the other. An msbuild step or analyzer script that runs at build time could be made to automate that, but I can't imagine we'll be changing defaults with any regularity anyway.

Reply:
Let me restate: I really like having a single file that contains all defaults. I do not like having defaults spread throughout the project.

Reply:
I got that and fully agree with the desire anyway, for multiple reasons, including simply for DRY.

That's what code generation or text templating is for, and why pretty much any project of similar complexity usually does have such things, either as an explicit step in the pre-design process when setting up the build environment (a la autoconf and the standard `configure` script it makes), or (more common and idiomatic for .net) at design time, via Roslyn. The goal is typically both a heavy helping of DRY and transparent automation of otherwise tedious or prone-to-be-forgotten tasks.

Pretty much every other option either results in multiple definitions (ew), incompatibilities/limitations, or are worse means of doing something standard.

For any future project-specific generators, I'd just suggest them being in their own repos, so they can have their output published right on github and automatically pulled into the Terminal.Gui project at design time (one of the most common ways of doing it, when it's not a generator meant for general public consumption), so you don't encounter difficulties with Visual Studio, like the project in the current solution.

Another automatic means of doing it that I've both seen occasionally and used myself a small number of times (mostly from other projects or before access to the Roslyn APIs was public) is having generators/templates/scripts that are needed for solution prep wired up as git hooks on post-checkout and also optionally as a pre-build step in the projects needing their output. That results in you having the files immediately upon checkout, and then a guarantee that, when you actually _build_ the project, the current source is used to update those files, if necessary.

The first solution is better, as you still get real-time generation for new code elements, and if set up as a submodule in this repo, it _feels_ like their source code is in this repo, even though it's not really.

_But_, I bet there are already generators for exactly this specific use already available on nuget. Maybe even something built in already. I haven't looked, in recent history. _That's_ the best solution for use with our custom configuration manager, if any are indeed available.

Reply:
All that said, I think that can wait til later anyway. Trim-compatibility is nice and a good goal, but it's not really a development blocker to finish other things up - more like just a release-blocker.